addons/web/models/models.py

```diff
@@ -52,6 +52,18 @@ def web_search_read(self, domain=None, fields=None, offset=0, limit=None, order=
         }
         """
         records = self.search_read(domain, fields, offset=offset, limit=limit, order=order)
+        return self._format_web_search_read_results(domain, records, offset, limit, count_limit)
+
+    @api.model
+    def unity_web_search_read(self, domain=None, specification=None, offset=0, limit=None, order=None, count_limit=None):
+        if specification is None:
+            specification = {'id': {}}
+
+        search = self.search(domain, offset=offset, limit=limit, order=order)
```

Comment:
search_fetch because you will need these data just after (it will save one query most of time).

Also can you add a small test to show that the number of query is only 1 (if `read` is not override and when there is not sub-record to read) ?